locust/runners.py

```diff
@@ -128,19 +128,24 @@ def spawn_locusts(self, spawn_count, hatch_rate, wait=False):
         
         def hatch():
             sleep_time = 1.0 / hatch_rate
+            hatch_count = 0
             while True:
                 if not bucket:
                     logger.info("All locusts hatched: %s (%i already running)" % (
                         ", ".join(["%s: %d" % (name, count) for name, count in occurrence_count.items()]), 
                         existing_count,
                     ))
+                    if not sorted([g.args[0].id for g in self.locusts]) == list(range(len(self.locusts))):
+                        logger.warning("Locust IDs are not consecutive.")
```

Comment:
When could this happen?

Reply:
This is mostly paranoia. If everything goes to plan, this should never happen.

New locusts get increasing IDs and that should happen every time.

Reply:
I think the check is fine. Just wanted to know if there were some case where this could happen that I had missed :).